### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by gauravdhawane14**
adding validation for user API but with RequestParam the @Valid annotation confused which property to do validate because I send user object through Postman.

can you check once and suggest me , how to resolve this issue.
Thank You !

### CODE REVIEW

File: SpringBootProject-jwt/src/main/java/com/springrest/controller/UserController.java [hunk 1]
```diff
@@ -37,6 +37,8 @@ public User add(@RequestParam("file") MultipartFile file, @RequestParam User use
 		
 		String path=filePathService.storeFile(file);
 		
+		System.out.println("System");
+
```
Comment (avinash827) [inline SpringBootProject-jwt/src/main/java/com/springrest/controller/UserController.java]: Please remove logs

File: SpringBootProject-jwt/pom.xml [hunk 2]
```diff
@@ -71,6 +71,18 @@
...
+			<artifactId>spring-boot-devtools</artifactId>
+			<scope>runtime</scope>
+		</dependency>
+
+
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-validation</artifactId>
+		</dependency>
 	</dependencies>
```
Comment (avinash827) [inline SpringBootProject-jwt/pom.xml:86]: Please remove extra space

File: SpringBootProject-jwt/src/main/java/com/springrest/controller/UserController.java [hunk 3]
```diff
@@ -27,90 +33,89 @@
...
 		return userService.removeUserById(id);
 	}
 
 	@PutMapping(path="/{id}")
 	public User editUserById(@PathVariable Integer id,@RequestParam("file") MultipartFile file,@RequestParam User user) {
-		
-        User user1=userService.getUserById(id);
-		
+
```
Comment (avinash827) [inline SpringBootProject-jwt/src/main/java/com/springrest/controller/UserController.java:90]: Please use proper naming conventions

File: SpringBootProject-jwt/src/main/java/com/springrest/controller/UserController.java [hunk 4]
```diff
@@ -27,90 +33,89 @@
...
+		User user1=userService.getUserById(id);
```
Comment (avinash827) [inline SpringBootProject-jwt/src/main/java/com/springrest/controller/UserController.java:91]: Please follow proper indentations , Please remove extra space

File: SpringBootProject-jwt/src/main/java/com/springrest/entity/Book.java [hunk 5]
```diff
@@ -15,6 +16,7 @@
 	@Id
 	@GeneratedValue
 	private Integer id;
+	@NotEmpty(message = "username is required")
 	private String name;
```
Comment (avinash827) [inline SpringBootProject-jwt/src/main/java/com/springrest/entity/Book.java:20]: This is not username , it's name property , please modify message

File: SpringBootProject-jwt/src/main/java/com/springrest/model/User.java [hunk 6]
```diff
@@ -18,18 +20,20 @@
...
 	private String username;
 	private String password;
 	private String firstName;
 	private String lastName;
-	@Column(unique = true)
+	//@Column(unique = true)
 	private String email;
-	@Column(unique = true)
+	//@Column(unique = true)
 	private Long phoneNo;
```
Comment (avinash827) [inline SpringBootProject-jwt/src/main/java/com/springrest/model/User.java:33]: Remove commented code

File: SpringBootProject-jwt/src/main/java/com/springrest/repository/ProfilePictureRepository.java [hunk 7]
```diff
@@ -1,9 +1,9 @@
...
-}
+//package com.springrest.repository;
+//
+//import org.springframework.data.jpa.repository.JpaRepository;
+//
+//import com.springrest.model.ProfilePitcture;
+//
+//public interface ProfilePictureRepository extends JpaRepository<ProfilePitcture, Integer> {
+//
+//}
```
Comment (avinash827) [inline SpringBootProject-jwt/src/main/java/com/springrest/repository/ProfilePictureRepository.java:9]: I see here all code are commented , if dont want this file please delete

**Review by avinash827 — CHANGES_REQUESTED** [review]

**Comment by gauravdhawane14** [comment]
Done with the conflicts resolution.

**Pull request merged**